src/script/components/MessagesList/Message/ContentMessage/asset/AudioAsset/AudioPlayButton/AudioPlayButton.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {ReactNode, useEffect, useState} from 'react';
+
+import {CloseIcon, PauseIcon, PlayIcon} from '@wireapp/react-ui-kit';
+
+import {useMessageFocusedTabIndex} from 'Components/MessagesList/Message/util';
+import {t} from 'Util/LocalizerUtil';
+
+import {playButtonStyles, wrapperStyles} from './AudioPlayButton.styles';
+
+import {AssetTransferState} from '../../../../../../../assets/AssetTransferState';
+
+export interface AudioPlayButtonProps {
+  mediaElement?: HTMLMediaElement;
+  onPause: () => void;
+  onPlay: () => void;
+  onCancel: () => void;
+  transferState: AssetTransferState;
+  isFocusable?: boolean;
+  isDisabled?: boolean;
+}
+
+export const AudioPlayButton = ({
+  mediaElement,
+  transferState,
+  onPlay,
+  onPause,
+  onCancel,
+  isFocusable = true,
+}: AudioPlayButtonProps) => {
+  const [isPlaying, setIsPlaying] = useState<boolean>(false);
+  const handlePlay = () => setIsPlaying(true);
+  const handlePause = () => setIsPlaying(false);
+
+  const messageFocusedTabIndex = useMessageFocusedTabIndex(isFocusable);
+
+  const isUploaded = transferState === AssetTransferState.UPLOADED;
+  const isDownloading = transferState === AssetTransferState.DOWNLOADING;
+  const isUploading = transferState === AssetTransferState.UPLOADING;
+
+  useEffect(() => {
+    if (mediaElement) {
+      mediaElement.addEventListener('playing', handlePlay);
+      mediaElement.addEventListener('pause', handlePause);
+    }
+    return () => {
+      if (mediaElement) {
+        mediaElement.removeEventListener('playing', handlePlay);
+        mediaElement.removeEventListener('pause', handlePause);
+      }
+    };
```

Comment:
I think early return will be fine there ? :)

```suggestion
    if (!mediaElement) {
      return;
    }
    
    mediaElement.addEventListener('playing', handlePlay);
    mediaElement.addEventListener('pause', handlePause);
      
    return () => {
      mediaElement.removeEventListener('playing', handlePlay);
      mediaElement.removeEventListener('pause', handlePause);
    };
```